### PR TITLE
Add ACK for disconnect

### PR DESCRIPTION
**Pull request by krish2718**
Pulled a dependent commit from #1140  will rebase once its merged.

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -139,6 +140,9 @@ enum net_event_wifi_cmd {
 
 #define NET_EVENT_WIFI_RAW_SCAN_RESULT                          \
 	(_NET_WIFI_EVENT | NET_EVENT_WIFI_CMD_RAW_SCAN_RESULT)
+
+#define NET_EVENT_WIFI_DISCONNECT_COMPLETE			\
```
Comment (markaj-nordic) [inline include/zephyr/net/wifi_mgmt.h:144]: Where is this one used? Is it that with this change we will still receive the NET_EVENT_WIFI_DISCONNECT_RESULT in the application with the difference being now we block the disconnect command until it really succeeds?

Comment (krish2718) [reply include/zephyr/net/wifi_mgmt.h:144]: No, this PR only adds the event, this is sent https://github.com/nrfconnect/sdk-hostap/pull/94/files#diff-ff43beace1c9eb2129211e73d03f9385752af39391aa954185b48b9891e1c9b4R305

Comment (markaj-nordic) [reply include/zephyr/net/wifi_mgmt.h:144]: Turned out the `NET_EVENT_WIFI_CMD_DISCONNECT_COMPLETE` was really not used :) Thanks for the change

**Review by rlubos — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_mgmt.c [hunk 2]
```diff
@@ -381,3 +381,15 @@ void wifi_mgmt_raise_raw_scan_result_event(struct net_if *iface,
...
 #endif /* CONFIG_WIFI_MGMT_RAW_SCAN_RESULTS */
+
+void wifi_mgmt_raise_disconnect_complete_event(struct net_if *iface,
+					       int status)
+{
+	struct wifi_status cnx_status = {
+		.status = status,
+	};
+
+	net_mgmt_event_notify_with_info(NET_EVENT_WIFI_CMD_DISCONNECT_COMPLETE,
```
Comment (kkasperczyk-no) [inline subsys/net/l2/wifi/wifi_mgmt.c:392]: Sorry for post merge review, but I verified it doesn't work. I can be fixed by replacing `NET_EVENT_WIFI_CMD_DISCONNECT_COMPLETE` with the `NET_EVENT_WIFI_DISCONNECT_COMPLETE`.

**Pull request merged**